### PR TITLE
Add query params to cluster

### PR DESCRIPTION
**Pull request by ashish-sharma09**
* Added config property for cluster uri within location client
* Added config property for authentication filter matching cluster uri
* Defined properties in environment specific yaml config files

### CODE REVIEW

File: pipe-http-server-cloud/src/integration/groovy/IdentityIssueTokenClientIntegrationSpec.groovy [hunk 1]
```diff
@@ -39,6 +39,9 @@ class IdentityIssueTokenClientIntegrationSpec extends Specification {
             .properties(
                 parseYamlConfig(
                 """
+                location:
+                  url:                "location_base_path"
+                  clusters.get.path.filter.pattern: "some/filter/pattern"
```
Comment (adam-yorwerth) [inline pipe-http-server-cloud/src/integration/groovy/IdentityIssueTokenClientIntegrationSpec.groovy:44]: I think indenting the values here makes it a bit unreadable, and it's inconsistent with some other tests (about 50% indent, 50% don't). If we can agree that no indent is better then perhaps we should change it?

File: pipe-http-server-cloud/src/integration/groovy/IdentityTokenValidatorIntegrationSpec.groovy [hunk 2]
```diff
@@ -62,6 +65,8 @@ class IdentityTokenValidatorIntegrationSpec extends Specification {
                 micronaut.security.token.jwt.bearer.enabled: true
                 micronaut.caches.identity-cache.expire-after-write: ${CACHE_EXPIRY_SECONDS}s
                 compression.threshold-in-bytes: 1024
+                location:
+                    clusters.get.path.filter.pattern: $LOCATION_CLUSTER_PATH_FILTER_PATTERN
```
Comment (adam-yorwerth) [inline pipe-http-server-cloud/src/integration/groovy/IdentityTokenValidatorIntegrationSpec.groovy:69]: Can all be one line `location.clusters.get...`

File: pipe-http-server-cloud/src/integration/groovy/IdentityIssueTokenClientIntegrationSpec.groovy (hunk 1 above)
Comment (sebastianlavender) [reply pipe-http-server-cloud/src/integration/groovy/IdentityIssueTokenClientIntegrationSpec.groovy:44]: I think indenting is fine with this version of loading the config properties and having tested both it looks better with an indent. I think the big issue here is the fact that we load the properties in multiple ways and I think we should standardise around one way. This should be picked up in another PR to keep this one as small as possible

**Review by LorenzoCianciaruso — APPROVED** [review]
LGTM

**Pull request merged**